oai_agents/common/checked_model_name_handler.py

```diff
@@ -0,0 +1,72 @@
+from oai_agents.common.tags import KeyCheckpoints
+import re
+from pathlib import Path
+from typing import Optional, List, Union
+
+class CheckedModelNameHandler:
+    def __init__(self):
+        """
+        Initializes the CheckedModelNameHandler with default prefix and reward substring.
+        """
+        self.prefix = KeyCheckpoints.CHECKED_MODEL_PREFIX
+        self.reward_substr = KeyCheckpoints.REWARD_SUBSTR
+        self.pattern = re.compile(f"^{re.escape(self.prefix)}(\\d+)(?:{re.escape(self.reward_substr)}[\\d.]+)?$")
+
+    def generate_tag(self, id: int, mean_reward: Optional[float] = None) -> str:
+        """
+        Generate a checked model name based on the given id and mean reward.
+
+        :param id: The identifier for the model, used as a numeric suffix.
+        :param mean_reward: Optional mean reward to include in the model name, required for ids greater than 0.
+        :return: A string representing the generated checked model name.
+        :raises ValueError: If id is negative or if mean_reward is not provided for ids greater than 0.
+        """
+        if id < 0:
+            raise ValueError("ID must be a non-negative integer.")
+
+        if id == 0:
+            return f"{self.prefix}{id}"
+
+        if mean_reward is None:
+            raise ValueError("Mean reward must be provided for IDs greater than 0.")
+
+        return f"{self.prefix}{id}{self.reward_substr}{mean_reward}"
+
+    def is_valid_checked_tag(self, tag: str) -> bool:
+        """
+        Check if a tag name matches the required pattern for checked models.
+
+        :param tag: The tag name to validate.
+        :return: True if the tag name matches the pattern; otherwise, False.
+        """
+        return bool(self.pattern.match(tag))
+
+    def get_all_checked_tags(self, path: Union[Path, None] = None) -> List[str]:
+        """
+        Retrieve all valid checked model tags (subdirectories) under the specified path that match the pattern.
+
+        :param path: The directory path to search for valid checked model tags. Can be a Path object or None.
+        :return: A list of valid checked model tag names.
+        :raises ValueError: If the path is None.
```

Comment:
You don't need to mention the potential errors it will raise in the function description

Reply:
will delete them!